### PR TITLE
[Storage] Support Cold tier in create/copy blob

### PR DESCRIPTION
**Pull request by blueww**
## Description

Besides upgrade SDK, this PR only add a value to PSArgumentCompleter, and update help. 
Doesn't include any code logic change.

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Storage/Storage/Blob/Cmdlet/CopyAzureStorageBlob.cs [hunk 1]
```diff
@@ -99,9 +99,9 @@ public string SrcContainer
         [ValidateSet("Block", "Page", "Append", IgnoreCase = true)]
         public string DestBlobType { get; set; }
 
-        [Parameter(HelpMessage = "Block Blob Tier, valid values are Hot/Cool/Archive. See detail in https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blob-storage-tiers", Mandatory = false)]
+        [Parameter(HelpMessage = "Block Blob Tier, valid values are Hot/Cool/Archive/Cold. See detail in https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blob-storage-tiers", Mandatory = false)]
```
Comment (BethanyZhou) [inline src/Storage/Storage/Blob/Cmdlet/CopyAzureStorageBlob.cs:102]:
```suggestion
        [Parameter(HelpMessage = "Block Blob Tier, valid values are Hot/Cool/Archive/Cold. See detail in https://docs.microsoft.com/azure/storage/blobs/storage-blob-storage-tiers", Mandatory = false)]
```
Remove culture localization information from url if possbile

File: src/Storage/Storage.Management/help/Start-AzStorageBlobCopy.md [hunk 2]
```diff
@@ -29,7 +29,7 @@ Start-AzStorageBlobCopy [-SrcBlob] <String> -SrcContainer <String> -DestContaine
 Start-AzStorageBlobCopy -CloudBlob <CloudBlob> [-BlobBaseClient <BlobBaseClient>] -DestContainer <String>
  [-DestBlob <String>] [-PremiumPageBlobTier <PremiumPageBlobTier>] [-StandardBlobTier <String>]
  [-RehydratePriority <RehydratePriority>] [-Tag <Hashtable>] [-Context <IStorageContext>]
- [-DestContext <IStorageContext>] [-DestTagCondition <String>] [-TagCondition <String>] [-Force]
+ [-DestContext <IStorageContext>] [-DestTagCondition <String>] [-TagCondition <String>] [-Force] [-AsJob]
```
Comment (BethanyZhou) [inline src/Storage/Storage.Management/help/Start-AzStorageBlobCopy.md:32]: https://github.com/Azure/azure-powershell/blob/b7e188390754a0a5c4f4323912e0b86374f8a691/src/Storage/Storage/Blob/Cmdlet/StartAzureStorageBlobCopy.cs#L306
I'm confused by -AsJob here, could you explain further?

**Review by BethanyZhou — CHANGES_REQUESTED** [review]

File: src/Storage/Storage.Management/help/Start-AzStorageBlobCopy.md (hunk 2 above)
Comment (blueww) [reply src/Storage/Storage.Management/help/Start-AzStorageBlobCopy.md:32]: The cmdlet really support -asjob before,  https://github.com/Azure/azure-powershell/blob/919880fa25aa35539f6308bbec224e4d9a8e0484/src/Storage/Storage/Blob/Cmdlet/StartAzureStorageBlobCopy.cs#L306

But it looks the help is not updated. So when update help, the cmdlet is added to help.

Comment (BethanyZhou) [reply src/Storage/Storage.Management/help/Start-AzStorageBlobCopy.md:32]: My point is seems like we commented AsJob in this command

File: src/Storage/Storage/Blob/Cmdlet/SetAzureStorageBlobContent.cs [hunk 3]
```diff
@@ -179,9 +179,9 @@ public PremiumPageBlobTier PremiumPageBlobTier
 
         private PremiumPageBlobTier? pageBlobTier = null;
 
-        [Parameter(HelpMessage = "Block Blob Tier, valid values are Hot/Cool/Archive. See detail in https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blob-storage-tiers", Mandatory = false)]
+        [Parameter(HelpMessage = "Block Blob Tier, valid values are Hot/Cool/Archive/Cold. See detail in https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blob-storage-tiers", Mandatory = false)]
```
Comment (BethanyZhou) [inline src/Storage/Storage/Blob/Cmdlet/SetAzureStorageBlobContent.cs:182]:
```suggestion
        [Parameter(HelpMessage = "Block Blob Tier, valid values are Hot/Cool/Archive/Cold. See detail in https://docs.microsoft.com/azure/storage/blobs/storage-blob-storage-tiers", Mandatory = false)]
```

File: src/Storage/Storage/Blob/Cmdlet/StartAzureStorageBlobCopy.cs [hunk 4]
```diff
@@ -210,9 +210,9 @@ public PremiumPageBlobTier PremiumPageBlobTier
         }
         private PremiumPageBlobTier? pageBlobTier = null;
 
-        [Parameter(HelpMessage = "Block Blob Tier, valid values are Hot/Cool/Archive. See detail in https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blob-storage-tiers", Mandatory = false)]
+        [Parameter(HelpMessage = "Block Blob Tier, valid values are Hot/Cool/Archive/Cold. See detail in https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blob-storage-tiers", Mandatory = false)]
```
Comment (BethanyZhou) [inline src/Storage/Storage/Blob/Cmdlet/StartAzureStorageBlobCopy.cs:213]:
```suggestion
        [Parameter(HelpMessage = "Block Blob Tier, valid values are Hot/Cool/Archive/Cold. See detail in https://docs.microsoft.com/azure/storage/blobs/storage-blob-storage-tiers", Mandatory = false)]
```

File: src/Storage/Storage.Management/help/Copy-AzStorageBlob.md [hunk 5]
```diff
@@ -318,7 +318,7 @@ Accept wildcard characters: False
 ```
 
 ### -StandardBlobTier
-Block Blob Tier, valid values are Hot/Cool/Archive.
+Block Blob Tier, valid values are Hot/Cool/Archive/Cold.
 See detail in https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blob-storage-tiers
```
Comment (BethanyZhou) [inline src/Storage/Storage.Management/help/Copy-AzStorageBlob.md:322]:
```suggestion
See detail in https://docs.microsoft.com/azure/storage/blobs/storage-blob-storage-tiers
```

**Review by BethanyZhou — CHANGES_REQUESTED** [review]

File: src/Storage/Storage/Blob/Cmdlet/SetAzureStorageBlobContent.cs (hunk 3 above)
Comment (blueww) [reply src/Storage/Storage/Blob/Cmdlet/SetAzureStorageBlobContent.cs]: Fixed

File: src/Storage/Storage/Blob/Cmdlet/StartAzureStorageBlobCopy.cs (hunk 4 above)
Comment (blueww) [reply src/Storage/Storage/Blob/Cmdlet/StartAzureStorageBlobCopy.cs]: Fixed

File: src/Storage/Storage.Management/help/Copy-AzStorageBlob.md (hunk 5 above)
Comment (blueww) [reply src/Storage/Storage.Management/help/Copy-AzStorageBlob.md]: Fixed

File: src/Storage/Storage.Management/help/Start-AzStorageBlobCopy.md (hunk 2 above)
Comment (blueww) [reply src/Storage/Storage.Management/help/Start-AzStorageBlobCopy.md:32]: I think we should leave -Asjob in the cmdlet, and update help to make it aligned.

Comment (blueww) [reply src/Storage/Storage.Management/help/Start-AzStorageBlobCopy.md]: @BethanyZhou 

You are right. I have removed the -asjob change from start-AzStorageBlobCopy help.
It's interesting why this parameter is generated when I update help last time, but it's removed when I update the help again.

Thanks for the catching!

**Review by BethanyZhou — APPROVED** [review]
LGTM

**Pull request merged**